### PR TITLE
[Ruby] Fixed keybind for interpolation and improved scopes.

### LINKED ISSUE #1458: [Ruby] Autocomplete within string interpolation doesn't complete word
When interpolating an existing variable into a string via `"#{}"`, the autocomplete dropdown shows up, but pressing `Tab` doesn't autocomplete the word. Instead, the cursor jumps to the outside of the right bracket.

Gif to demonstrate:

string_interp

### PR DESCRIPTION
**Pull request by Thom1729**
Fixes #1458. Uses interpolation scopes based on Python. See #1432 for further discussion.

### CODE REVIEW

File: Ruby/Ruby.sublime-syntax [hunk 1]
```diff
@@ -1252,17 +1252,14 @@ contexts:
...
 
   interpolated-ruby:
-    - match: '(#\{)(\})'
-      scope: punctuation.section.embedded.ruby
-      captures:
-        1: source.ruby.embedded.source
-        2: source.ruby.embedded.source.empty
     - match: '#\{'
-      scope: punctuation.section.embedded.ruby
+      scope: punctuation.definition.interpolation.begin.ruby
```
Comment (FichteFoll) [inline Ruby/Ruby.sublime-syntax]: Via https://github.com/sublimehq/Packages/pull/1432#issuecomment-366750733, I believe `punctuation.section.interpolation.*` should be used.

If this is not currently the case with Python, it should be adjusted imo.

Comment (wbond) [reply Ruby/Ruby.sublime-syntax]: @Thom1729 this should be

```suggestion
      scope: punctuation.section.interpolation.begin.ruby
```

File: Ruby/Ruby.sublime-syntax [hunk 2]
```diff
@@ -1252,17 +1252,14 @@ contexts:
...
       push:
-        - meta_scope: source.ruby.embedded.source
+        - clear_scopes: 1
+        - meta_scope: meta.interpolation
+        - meta_content_scope: source.ruby.embedded
         - match: '\}'
-          scope: punctuation.section.embedded.ruby
+          scope: punctuation.definition.interpolation.end.ruby
```
Comment (wbond) [inline Ruby/Ruby.sublime-syntax]: This should be

```suggestion
          scope: punctuation.section.interpolation.end.ruby
```

File: Ruby/syntax_test_ruby.rb [hunk 3]
```diff
@@ -20,7 +20,8 @@
 
 class_eval <<-RUBY, __FILE__, __LINE__ + 1
   def #{sym}(*args, &block)
-#     ^^ punctuation.section.embedded
+#     ^^ punctuation.definition.interpolation.begin
```
Comment (wbond) [inline Ruby/syntax_test_ruby.rb]: This should be

```suggestion
#     ^^ punctuation.section.interpolation.begin
```

File: Ruby/syntax_test_ruby.rb [hunk 4]
```diff
@@ -20,7 +20,8 @@
...
+#          ^ punctuation.definition.interpolation.end
```
Comment (wbond) [inline Ruby/syntax_test_ruby.rb]: This should be

```suggestion
#          ^ punctuation.section.interpolation.end
```

**Comment by Thom1729** [comment]
All done.

**Pull request merged**